src/microbe_stage/Microbe.cs

```diff
@@ -938,6 +947,51 @@ public List<(Compound Compound, Color Colour, Vector3 Target)> GetDetectedCompou
         return detections;
     }
 
+    /// <summary>
+    ///   Returns a list of tuples, representing all possible microbe targets. These are not all the
+    ///   other microbes that the microbe can smell; only the best candidate of each species.
+    /// </summary>
+    /// <param name="microbeSystem">MicrobeSystem to scan</param>
+    /// <returns>
+    ///   A list of tuples. Each tuple contains the type of species, the color of the line (if any needs to be drawn),
+    ///   and the location where the microbe is located.
+    /// </returns>
+    public List<(Microbe Microbe, Color Colour, Vector3 Target)> GetDetectedSpecies(MicrobeSystem microbeSystem)
+    {
+        HashSet<(Species Species, float Range, Color Colour)> collectedUniqueSpeciesDetections;
+
+        // Colony lead cell uses all the chemoreceptors in the colony to make them all work
+        if (Colony != null && Colony.Master == this)
+        {
+            collectedUniqueSpeciesDetections =
+                new HashSet<(Species Species, float Range, Color Colour)>();
+
+            foreach (var colonyMicrobe in Colony.ColonyMembers)
+            {
+                collectedUniqueSpeciesDetections.UnionWith(colonyMicrobe.activeSpeciesDetections);
+            }
+        }
+        else
+        {
+            collectedUniqueSpeciesDetections = activeSpeciesDetections;
+        }
+
+        var detections = new List<(Microbe Microbe, Color Colour, Vector3 Target)>();
+        var position = GlobalTranslation;
+
+        foreach (var (species, range, colour) in collectedUniqueSpeciesDetections)
+        {
+            var tuple = microbeSystem.FindSpeciesNearPoint(position, species, range);
```

Comment:
Oh and this is called here in a recursive loop. This is absolutely going to wreck performance if you have a colony of 50 cells each with multiple chemoreceptor and hundreds of cells in the game, the full list of microbes in the world is going to be rebuilt so many times in that situation.